src/Scheduling/MassTransit.QuartzIntegration/QuartzIntegration/JobDataMessageContext.cs

```diff
@@ -154,6 +154,9 @@ Headers GetHeaders()
             if (_jobDataMap.TryGetValue("TokenId", out var tokenId))
                 headers.Set(MessageHeaders.SchedulingTokenId, tokenId);
 
+            headers.Set(MessageHeaders.Quartz.ScheduleId, _executionContext.JobDetail.Key.Name);
```

Comment:
These should only be set if they aren't empty, you can add the appropriate string.IsNullOrWhitespace checK?

Reply:
Both Name and Group is not nullable string properties in JobKey. They always exist.

Reply:
Not nullable doesn't mean it isn't empty. Don't put too much faith in the nullable attribute of a string. Net6 has only been out for a year, and quartz has existed for 10x that.

Reply:
Done.